crates/bevy_ecs/src/archetype.rs

```diff
@@ -153,16 +176,18 @@ impl Edges {
 }
 
 pub struct ArchetypeEntity {
```

Comment:
Docs needed here: I *think* this is an entity that's associated with a specific archetype?

Reply:
I'm actually a bit iffy on this one. If #6681 gets merged, this actually has zero business being pub anymore, as the table row is fetchable via `EntityLocation` instead. It's realistically a type that is made to optimize the layout for sparse iteration.

Reply:
Added temp placeholder docs until this either gets hidden or we decide otherwise.